domain/consensus/processes/dagtraversalmanager/dagtraversalmanager_test.go

```diff
@@ -111,3 +212,75 @@ func TestLowestChainBlockAboveOrEqualToBlueScore(t *testing.T) {
 		checkExpectedBlock(tipHash, 10, blueScore11BlockHash)
 	})
 }
+
+func createAChainDAG(genesisHash *externalapi.DomainHash, tc testapi.TestConsensus) (*externalapi.DomainHash, error) {
+	block := genesisHash
+	var err error
+	for i := 0; i < sizeOfTheChainThatCommonToAllGraphs; i++ {
+		block, _, err = tc.AddBlock([]*externalapi.DomainHash{block}, nil, nil)
+		if err != nil {
+			return nil, err
+		}
+	}
+	return block, nil
+}
+
+func createADAGTwoChildrenWithSameSelectedParent(genesisHash *externalapi.DomainHash, tc testapi.TestConsensus) (*externalapi.DomainHash, *externalapi.DomainHash, error) {
+	block := genesisHash
+	var err error
+	for i := 0; i < sizeOfTheChainThatCommonToAllGraphs; i++ {
+		block, _, err = tc.AddBlock([]*externalapi.DomainHash{block}, nil, nil)
+		if err != nil {
+			return nil, nil, err
+		}
+	}
+	firstChild, _, err := tc.AddBlock([]*externalapi.DomainHash{block}, nil, nil)
+	if err != nil {
+		return nil, nil, err
+	}
+	secondChild, _, err := tc.AddBlock([]*externalapi.DomainHash{block}, nil, nil)
+	if err != nil {
+		return nil, nil, err
+	}
+	return firstChild, secondChild, nil
+}
+
+func createADAGWithTwoDifferentChains(genesisHash *externalapi.DomainHash, tc testapi.TestConsensus) (*externalapi.DomainHash, *externalapi.DomainHash, error) {
+	const sizeOfTheFirstChildSubChainDAG int = 3
+	const sizeOfTheSecondChildSubChainDAG int = 2
+	block := genesisHash
+	var err error
+	for i := 0; i < sizeOfTheChainThatCommonToAllGraphs; i++ {
+		block, _, err = tc.AddBlock([]*externalapi.DomainHash{block}, nil, nil)
+		if err != nil {
+			return nil, nil, err
+		}
+	}
+	firstChainHash, _, err := tc.AddBlock([]*externalapi.DomainHash{block}, nil, nil)
```

Comment:
These are not hashes of the chain - they are hashes of the chain tips.
It could be nice that this distinction is reflected in the name (a.k.a. rename to firstChainTipHash, secondChainTipHash)